### PR TITLE
Cache translated imports in PEModuleBuilder

### LINKED ISSUE #60045: ExternAliases4 test throws if `GetEmitDiagnostics` call added
```
        [Fact]
        public void ExternAliases4()
        {
            var src1 = @"
namespace N
{
    public class C { }
}";
            var dummyCompilation = CreateCompilation(src1, assemblyName: "A", options: TestOptions.DebugDll);

            var src2 = @"
namespace M
{
    extern alias A;
    using A::N;
    
    public class D
    {
        public C P
        {
            get { return new C(); }
            set { }
        }
    }
}";
            var compilation = CreateCompilation(src2,
                assemblyName: GetUniqueName(),
                options: TestOptions.DebugDll,
                references: new[]
                {
                    new CSharpCompilationReference(dummyCompilation, ImmutableArray.Create("A", "A")),
                });

            compilation.VerifyDiagnostics();
            compilation.GetEmitDiagnostics(); // System.InvalidOperationException : This program location is thought to be unreachable.
            compilation.VerifyEmitDiagnostics();
        }
```

```
Message: 
System.InvalidOperationException : This program location is thought to be unreachable.

  Stack Trace: 
PdbWriter.GetAssemblyReferenceAlias(IAssemblyReference assembly, HashSet`1 declaredExternAliases) line 442
PdbWriter.TryEncodeImport(UsedNamespaceOrType import, HashSet`1 declaredExternAliasesOpt, Boolean isProjectLevel) line 320
PdbWriter.DefineNamespaceScopes(IMethodBody methodBody) line 192
PdbWriter.SerializeDebugInfo(IMethodBody methodBody, StandaloneSignatureHandle localSignatureHandleOpt, CustomDebugInfoWriter customDebugInfoWriter) line 124
MetadataWriter.SerializeMethodBodies(BlobBuilder ilBuilder, PdbWriter nativePdbWriterOpt, Blob& mvidStringFixup) line 2907
MetadataWriter.BuildMetadataAndIL(PdbWriter nativePdbWriterOpt, BlobBuilder ilBuilder, BlobBuilder mappedFieldDataBuilder, BlobBuilder managedResourceDataBuilder, Blob& mvidFixup, Blob& mvidStringFixup) line 1823
PeWriter.WritePeToStream(EmitContext context, CommonMessageProvider messageProvider, Func`1 getPeStream, Func`1 getPortablePdbStreamOpt, PdbWriter nativePdbWriterOpt, String pdbPathOpt, Boolean metadataOnly, Boolean isDeterministic, Boolean emitTestCoverageData, Nullable`1 privateKeyOpt, CancellationToken cancellationToken) line 71
Compilation.SerializePeToStream(CommonPEModuleBuilder moduleBeingBuilt, DiagnosticBag metadataDiagnostics, CommonMessageProvider messageProvider, Func`1 getPeStream, Func`1 getMetadataPeStreamOpt, Func`1 getPortablePdbStreamOpt, PdbWriter nativePdbWriterOpt, String pdbPathOpt, RebuildData rebuildData, Boolean metadataOnly, Boolean includePrivateMembers, Boolean isDeterministic, Boolean emitTestCoverageData, Nullable`1 privateKeyOpt, CancellationToken cancellationToken) line 3250
Compilation.SerializeToPeStream(CommonPEModuleBuilder moduleBeingBuilt, EmitStreamProvider peStreamProvider, EmitStreamProvider metadataPEStreamProvider, EmitStreamProvider pdbStreamProvider, RebuildData rebuildData, Func`2 testSymWriterFactory, DiagnosticBag diagnostics, EmitOptions emitOptions, Nullable`1 privateKeyOpt, CancellationToken cancellationToken) line 3147
Compilation.Emit(Stream peStream, Stream metadataPEStream, Stream pdbStream, Stream xmlDocumentationStream, Stream win32Resources, IEnumerable`1 manifestResources, EmitOptions options, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, RebuildData rebuildData, CompilationTestData testData, CancellationToken cancellationToken) line 2886
Compilation.Emit(Stream peStream, Stream pdbStream, Stream xmlDocumentationStream, Stream win32Resources, IEnumerable`1 manifestResources, EmitOptions options, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, Stream metadataPEStream, RebuildData rebuildData, CancellationToken cancellationToken) line 2778
Compilation.Emit(Stream peStream, Stream pdbStream, Stream xmlDocumentationStream, Stream win32Resources, IEnumerable`1 manifestResources, EmitOptions options, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, Stream metadataPEStream, CancellationToken cancellationToken) line 2666
DiagnosticExtensions.GetEmitDiagnostics[TCompilation](TCompilation c, EmitOptions options, IEnumerable`1 manifestResources) line 351
DiagnosticExtensions.VerifyEmitDiagnostics[TCompilation](TCompilation c, EmitOptions options, DiagnosticDescription[] expected) line 340
DiagnosticExtensions.VerifyEmitDiagnostics[TCompilation](TCompilation c, DiagnosticDescription[] expected) line 357
PDBUsingTests.ExternAliases4() line 607
```

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #60045.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/PDB/PDBUsingTests.cs [hunk 1]
```diff
@@ -606,6 +606,7 @@ public C P
                 });
 
             compilation.VerifyDiagnostics();
+            compilation.GetEmitDiagnostics();
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Emit/PDB/PDBUsingTests.cs:609]: Consider making this method a regular `Fact` now that the UsedAssemblies issue is fixed.

**Review by 333fred — COMMENTED** [review]
Can we add a test that would fail in VB without this change?

**Review by 333fred — APPROVED** [review]

**Comment by AlekseyTs** [comment]
> Can we add a test that would fail in VB without this change?

I am comfortable making the change for VB without specific failure case.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review

**Review by cston — APPROVED** [review]

**Pull request merged**